x-pack/plugins/global_search_bar/public/components/search_bar.tsx

```diff
@@ -46,15 +49,25 @@ const EmptyMessage = () => (
   </EuiFlexGroup>
 );
 
+const GLOBAL_SEARCH_BAR_VISIBLE_KEY = 'GLOBAL_SEARCH_BAR_VISIBLE' as const;
+
 export const SearchBar: FC<SearchBarProps> = ({
   globalSearch,
   taggingApi,
   navigateToUrl,
   trackUiMetric,
+  chromeStyle$,
   ...props
 }) => {
   const isMounted = useMountedState();
   const { euiTheme } = useEuiTheme();
+  const chromeStyle = useObservable(chromeStyle$);
+
+  // These hooks are used when on chromeStyle set to 'project'
+  const [isVisible, setIsVisible] = useLocalStorage(GLOBAL_SEARCH_BAR_VISIBLE_KEY, false);
```

Comment:
I found this a bit weird that we keep this state inside the storage. It seems that usually, this kind of state is just in memory.

Reply:
I think there is benefit to preserving the selection, since the search bar used to be always visible and now it takes a click to show it. If it's ok, I'll go ahead with it this way for now, and bring it up for discussion in the [project document](https://docs.google.com/document/d/1Z-AkRoagS6JMVN5YHZpjgQr0sYGFXBTNS9SHsbLPfGQ/edit#heading=h.rowzutqq7ldp).

Reply:
This follows the setting for the expand/collapse of the side navigation.

You might be right that it's better to not store this. I would prefer to make this a simple PR and not add anything that isn't really necessary. If there is friction, we can store the state with a tiny change.

Pushed 82c5e60